drivers/cisco/ise/network_access.cr

```diff
@@ -199,21 +168,46 @@ class Cisco::Ise::Guests < PlaceOS::Driver
     JSON.parse(response.body)
   end
 
-  def update_internal_user_password_by_email(email : String, password : String)
+  def update_internal_user_password_by_email(email : String, password : String? = nil)
+    password ||= generate_password
     internal_user = get_internal_user_by_email(email)
 
     update_internal_user_password_by_id(internal_user.id.to_s, password)
   end
 
-  # Will be 9 characters in length until 2081-08-05 10:16:46.208000000 UTC
-  # when it will increase to 10
-  private def genererate_username(firstname, lastname)
-    "#{firstname[0].downcase}#{lastname[0].downcase}#{Time.utc.to_unix_ms.to_s(62)}"
-  end
-
-  # Will be 9 characters in length until 2081-08-05 10:16:46.208000000 UTC
-  # when it will increase to 10
-  private def genererate_password(firstname, lastname)
-    "P!#{lastname[0].downcase}#{firstname[0].downcase}#{Time.utc.to_unix_ms.to_s(31)}"
+  # Todo, when ISE doesn't return 401 for Guest related api calls
+  # def create_guest (...)
+  #   # sms_service_provider ||= @sms_service_provider
+  #   # guest_type ||= @guest_type
+  #   # portal_id ||= @portal_id
+
+  #   # time_object = Time.unix(event_start).in(@timezone)
+  #   # from_date = time_object.at_beginning_of_day.to_s(TIME_FORMAT)
+  #   # to_date = time_object.at_end_of_day.to_s(TIME_FORMAT)
+
+  #   # If company_name isn't passed
+  #   # Hackily grab a company name from the attendee's email (we may be able to grab this from the signal if possible)
+  #   # company_name ||= attendee_email.split('@')[1].split('.')[0].capitalize
+
+  # These custom attributes and any custom attribute needs to be predefined
+  # in the ISE GUI.
+  # custom_attributes = {
+  #   "fromDate"           => from_date,
+  #   "toDate"             => to_date,
+  #   "location"           => @location.to_s,
+  #   "companyName"        => company_name,
+  #   "phoneNumber"        => phone_number,
+  #   "smsServiceProvider" => sms_service_provider.to_s,
+  #   "guestType"          => guest_type,
+  #   "portalId"           => portal_id,
+  # } of String => String
+
+  # custom_attributes.merge!(@custom_data)
+  # end
+
+  # Will be lowercase letters and numbers
```

Comment:
This might be an issue since it needs one uppercase character and a symbol character to be present in the password